kernel/declarations.h

```diff
@@ -0,0 +1,361 @@
+#ifndef DECL
+#define DECL
+
+// Types
+// ----------------------------------------------------------------------------
+
+typedef unsigned int   uint;
+typedef unsigned short ushort;
+typedef unsigned char  uchar;
+
+typedef unsigned char uint8;
+typedef unsigned short uint16;
+typedef unsigned int  uint32;
+typedef unsigned long uint64;
+
+typedef uint64 pde_t;
+
+// System
+// ----------------------------------------------------------------------------
+
+// System Parameters
+#define NREG        31                  // Number of Registers
+#define BSIZE       1024                // Disk Block Size
+#define PGSIZE      4096                // Bytes per Page
+#define PGSHIFT     12                  // Bits of Offset within a Page
+#define FSSIZE    1000                // size of disk in blocks
+#define CPUID       0
+#define MAXOPBLOCKS 10                  // max # of blocks any op writes
+
+// Machine Status Register, mstatus
```

Comment:
You can write one-two words for every machine status register

Reply:
I'll do it